nixos/modules/services/x11/desktop-managers/plasma5.nix

```diff
@@ -37,32 +37,6 @@ in
 
   config = mkMerge [
     (mkIf (xcfg.enable && cfg.enable) {
-      services.xserver.desktopManager.session = singleton {
-        name = "plasma5";
-        bgSupport = true;
-        start = ''
-          # Load PulseAudio module for routing support.
-          # See http://colin.guthr.ie/2009/10/so-how-does-the-kde-pulseaudio-support-work-anyway/
-          ${optionalString config.hardware.pulseaudio.enable ''
```

Comment:
Are all the extra things in here not needed?

If not you can maybe do something like https://github.com/NixOS/nixpkgs/blob/1a66e1dd276940e9666ff5fee1f1a739f3a55736/nixos/modules/services/x11/desktop-managers/pantheon.nix#L80

Reply:
Yeah that's a good idea actually. Those will still be run with upstream sessions?

Reply:
I think we can drop this, if you look here: https://bugs.freedesktop.org/show_bug.cgi?id=95104

Reply:
I tested switchting audio sources and worked without any problems for me.